src/scripts/sync_synonym.py

```diff
@@ -54,19 +57,51 @@ def _query_synonyms(ids: List[CURIE], db: SqlImplementation) -> pd.DataFrame:
     return df
 
 
-def _filter_format_save(
+# todo: DRYify? this and _merge_synonym_types() share similar logic
+def _add_synonym_type_inferences(row: pd.Series) -> str:
+    """Append additional synonym_type(s) based on logical rules."""
+    syn: str = row['synonym']
+    syn_type_str: str = row['synonym_type']
+    types: Set[URI] = set(syn_type_str.split('|') if syn_type_str else [])
+
+    # Acronym: uppercase, no numbers, no whitespace, <10 chars
+    if syn.isupper() and syn.isalpha() and not any(char.isspace() for char in syn) and len(syn) < 10:
+        types.add('http://purl.obolibrary.org/obo/mondo#ABBREVIATION')
+
+    return '|'.join(types)
+
+
+def _merge_synonym_types(row: pd.Series) -> str:
+    """Merge Mondo synonym_types into those obtained from the source so that Mondo types are not overwritten."""
```

Comment:
## Relationship of `synonym_type` to `synonym_type_mondo`
- [ ] @matentzn @twhetzel I think you guys may want to pay more attention to this topic. Want to make sure what I did was correct.

`synonym_type_mondo` shows all synonym types currently declared on the synonym from Mondo.

`synonym_type` is used to import new synonyms into Mondo.
However, I was worried. I didn't get time to check, but I think that what is in this column will potentially overwrite what's already in Mondo.

For example, say that a synonym in Mondo has `syn_type_a`, but the synonym sync finds that it also has a new type `syn_type_b`. I think that if the `synonym_type` field is only set to 'b', then 'a' will get overwritten. So what I've done is merged them. Now in this case, `synonym_type` gets set to `syn_type_a|syn_type_b`.
I haven't had a chance yet to test this 'overwritten' behavior, but if I am wrong, I can change this to not merge the two.

<details><summary>Additional granular synonym_type fields I considered</summary>
<p>

I'm not a curator so I can't say, but it might be useful for us to be able to distinguish the following cases:
- The synonym type from Mondo
- Declared: The synonym type declared in the source artefact
  - e.g. OMO:0003012 is declared in the DOID mirror
- Inferred (via synonym sync): New synonym type generated through the synonym sync logic
  - e.g. <10 chars, all caps, etc = acronym
- Created through a non-synonym-sync pipeline: e.g. `MONDO:GENERATED`, which is on several components. I think that the component creation pipeline makes some of these, but some of them also might get made during individual source pipelines (e.g. the OMIM pipeline in its own git repo)

</p>
</details>

Reply:
The source synonym type information and mondo synonym type information should not be merged into the `synonym_type` column. This column should only contain any source synonym type information that already exists in the source such as OMO:0003012 _or_ the inferred synonym type based on the abbreviation rule in the script.

Reply:
Sure, I'll change that.

I assume that means that we do not think that the ROBOT behavior here will be to overwrite the synonym type in Mondo, but to append to it.